code/components/extra-natives-five/src/VehicleExtraNatives.cpp

```diff
@@ -363,4 +367,110 @@ static HookFunction initFunction([]()
 	{
 		*reinterpret_cast<float *>(wheelAddr + WheelXOffsetOffset) = context.GetArgument<float>(2);
 	}));
+
+	static struct : jitasm::Frontend {
+		static bool ShouldSkipRepairFunc(fwEntity* VehPointer)
+		{
+
+			for (auto it = vehicles.begin(); it != vehicles.end(); it++) {
+
+				if (VehPointer == *it) {
+					return true;
+				}
+			}
+			return false;
+		}
+		virtual void InternalMain() override
+		{
+			//Restore registers from call and push registers to the stack to be preserved
+			mov(rcx, rbx);
+			push(rax);
+			push(rdx);
+
+			sub(rsp, 0x28);
+			mov(rax, (uintptr_t)ShouldSkipRepairFunc);
+			call(rax);
+			add(rsp, 0x28);
+
+			pop(rdx);
+			cmp(al, true); //al is 1 if repair func should be skipped
+			je("skiprepair");
+			pop(rax);
+			sub(rsp, 0x28);
+			AppendInstr(jitasm::InstrID::I_CALL, 0xFF, 0, jitasm::Imm8(2), qword_ptr[rax + 0x5D0]);
```

Comment:
could this perhaps be an overload added to jitasm.h directly, depending on the return type of qword_ptr[...]?